### ISSUE TITLE
Add Business Day arithmetic

### ISSUE THREAD

**Issue by choptastic** [choptastic/qdate#3]
Per the recommendation from @RomanShestakov, add business day arithmetic.

This will require either accepting a list of business days, or more likely, have a predicate function passed to it which would evaluate whether or not a date is a business day.

This would work with the `Modifier` in https://github.com/choptastic/qdate/issues/1 by adding three new date units: `business_day` and `business_week`, `business_month`
- `business_day` would check each day, only decrementing the counter if the day is actually a business day.
- `business_week` will increment like a normal week, and then continue incrementing each day until it finds the first business day. Example: Assume today is monday, and next monday is a holiday: 1 business week would be next tuesday, while 2 business weeks would just be two mondays from now.
- `business_month` will work exactly like `business_week`, except it increments the month.

In short: `business_week` and `business_month` would do valid-business-day checks only at the end, while `business_day` does one for each day.

**Comment by marutha** [comment]
+1 any updates on this?

**Comment by choptastic** [comment]
Not yet.  I haven't had the immediate need in my apps yet for it, so the priority of building it has fallen in lieu of more immediate needs.
